pytorch_lightning/utilities/cli.py

```diff
@@ -0,0 +1,274 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Type, Dict, Any
+from jsonargparse import ArgumentParser, ActionConfigFile
+from pytorch_lightning.trainer.trainer import Trainer
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.core.datamodule import LightningDataModule
+from pytorch_lightning.callbacks import Callback
+
+
+class LightningArgumentParser(ArgumentParser):
+    """Extension of jsonargparse's ArgumentParser for pytorch-lightning"""
+
+    def __init__(
+        self,
+        *args,
+        parse_as_dict: bool = True,
+        **kwargs
+    ):
+        """Initialize argument parser that supports configuration file input
+
+        For full details of accepted arguments see `ArgumentParser.__init__
+        <https://omni-us.github.io/jsonargparse/#jsonargparse.core.ArgumentParser.__init__>`_.
+        """
+        super().__init__(*args, parse_as_dict=parse_as_dict, **kwargs)
+        self.add_argument(
+            '--config',
+            action=ActionConfigFile,
+            help='Path to a configuration file in json or yaml format.'
+        )
+
+    def add_trainer_args(
```

Comment:
I have just pushed a change so that there is only one method to add arguments, now called `add_lightning_class_args` so that the purpose is clear in comparison to the methods of the base class. I also added some asserts to `LightningCLI` since by having a single add method people could make a mistake giving a class to the wrong argument so the asserts check this. @Borda see how it looks now.